nexus/internal/filetransfer/file_transfer_default.go

```diff
@@ -65,10 +66,21 @@ func (ft *DefaultFileTransfer) Upload(task *UploadTask) error {
 	return nil
 }
 
+func (d *DownloadTask) ensureDownloadRootDir() error {
+	baseDir := path.Dir(d.Path)
+	info, err := os.Stat(baseDir)
+	if err == nil && info.IsDir() {
+		return nil
+	}
+	return os.MkdirAll(baseDir, 0777)
+}
+
 // Download downloads a file from the server
 func (ft *DefaultFileTransfer) Download(task *DownloadTask) error {
-	ft.logger.Debug("default file transfer: downloading file", "path", task.Path, "url", task.Url)
-	// open the file for reading and defer closing it
+	// Skip downloading the file if it already exists
```

Comment:
Did you intend to add code for this here? If so, let's do that in the callsite instead.